sphinx/domains/std.py

```diff
@@ -451,17 +460,38 @@ def run(self) -> List[Node]:
             subnode = addnodes.production(rule)
             subnode['tokenname'] = name.strip()
             if subnode['tokenname']:
-                idname = nodes.make_id('grammar-token-%s' % subnode['tokenname'])
+                idname = 'grammar-token2-%s_%s' \
```

Comment:
I hesitate to add "2" suffix for new styled node_id. I feel it is a bit ugly.

Reply:
It is indeed, and is actually not needed as the ``_`` is enough that IDs will not clash. I am added a comment in the code to explain the details.